src/Bicep.Core/Semantics/Namespaces/ThisNamespaceType.cs

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Concurrent;
+using System.Collections.Immutable;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Features;
+using Bicep.Core.Intermediate;
+using Bicep.Core.Syntax;
+using Bicep.Core.TypeSystem;
+using Bicep.Core.TypeSystem.Providers;
+using Bicep.Core.TypeSystem.Types;
+
+namespace Bicep.Core.Semantics.Namespaces;
+
+public static class ThisNamespaceType
+{
+    // Note: self() function should be added to this namespace when it is implemented in bicep. 
+    public const string BuiltInName = "this";
+
+    public static NamespaceSettings Settings { get; } = new(
+        IsSingleton: true,
+        BicepExtensionName: BuiltInName,
+        ConfigurationType: null,
+        TemplateExtensionName: "This",
+        TemplateExtensionVersion: "1.0.0");
```

Comment:
[Nit]: Should this be "0.0.1" until the feature is no longer experimental? Not sure where or how this version number is used.